engines/libaio.c

```diff
@@ -123,7 +127,13 @@ static int fio_libaio_prep(struct thread_data *td, struct io_u *io_u)
 		if (o->nowait)
 			iocb->aio_rw_flags |= RWF_NOWAIT;
 	} else if (io_u->ddir == DDIR_WRITE) {
-		io_prep_pwrite(iocb, f->fd, io_u->xfer_buf, io_u->xfer_buflen, io_u->offset);
+		if ((td->o.zone_mode == ZONE_MODE_ZBD || td->o.zone_mode == ZONE_MODE_STRIDED) &&
+		     td->o.zone_append) {
+			io_prep_pwrite(iocb, f->fd, io_u->xfer_buf, io_u->xfer_buflen, io_u->zone_slba);
+			iocb->aio_lio_opcode = IO_CMD_ZONE_APPEND;
+		}
+		else
+			io_prep_pwrite(iocb, f->fd, io_u->xfer_buf, io_u->xfer_buflen, io_u->offset);
```

Comment:
If you can set io_u->offset to zone->start in zbd_adjust_block() for zone append, you would only need to set the opcode here. That is, if we define a new opcode... Back to first defining the kernel interface as it is not clear yet.

Reply:
New opcodes will be removed in the v2 patches and we will be using RWF_ZONE_APPEND flag as specified in the comments for the Kernel patches.
io_u->offset cannot be used due to the random map generator issue.

Reply:
And verify will not work either. Lots of things need to be documented about zone append use, in the commit messages and man page.

Reply:
verify will work fine, just as normal write (we are updating io_u->ipo->offset)

Reply:
The offset being written is used as data for write buffers snd checked on read, no ? So verify cannot work for zone append without modifications, I think.

Reply:
io_u->ipo->offset is updated as the offset returned from Linux kernel block layer in the fio_libaio_getevents()

io_u->offset is updated from ipo->offset at the time of verification in the get_next_verify() function of verify.c
This offset is used to perform read and verify the data written earlier.

Reply:
I do not get it... Will need to check the code again. My understanding is that since io_u->offset is part of the data written, one cannot make verify work since that offset may not be equal to the one where the data was actually written. So on read, you can get mismatches of offset in data with the offset being read.